src/main/java/net/sf/jabref/BibDatabaseContext.java

```diff
@@ -175,13 +175,14 @@ private String getFileDirectoryPath(String directoryName) {
         String dir = directoryName;
         // If this directory is relative, we try to interpret it as relative to
         // the file path of this BIB file:
-        if (!new File(dir).isAbsolute() && (getDatabaseFile() != null)) {
+        Optional<File> databaseFile = getDatabaseFile();
+        if (!new File(dir).isAbsolute() && databaseFile.isPresent()) {
```

Comment:
I am not quite sure about this one, but if you use the nio Paths methods, you could omit the check for absolute/relative directory.